test/unittest/unittest_nntrainer_layers.cpp

```diff
@@ -758,6 +758,151 @@ TEST(nntrainer_Pooling2D, forwarding_02_p) {
+  int status = ML_ERROR_NONE;
+  nntrainer::Pooling2DLayer layer;
+  std::vector<std::string> input_str;
+  nntrainer::TensorDim previous_dim;
+  previous_dim.setTensorDim("1:2:5:5");
+  layer.setInputDimension(previous_dim);
+
+  input_str.push_back("pooling = global_max");
+
+  status = layer.setProperty(input_str);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = layer.initialize(false);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  nntrainer::Tensor in(1, 2, 5, 5);
+  nntrainer::Tensor out;
+  std::ifstream file("test_1_goldenConv2DResult.out");
+  in.read(file);
+  file.close();
+  out = layer.forwarding(in, status);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  float *out_ptr;
+
+  float golden[2] = {7.8846731, 8.81525};
+
+  out_ptr = out.getData();
+
+  for (int i = 0; i < 2; ++i) {
+    EXPECT_FLOAT_EQ(out_ptr[i], golden[i]);
+  }
+}
+
+/**
+ * @brief Pooling 2D Layer
+ */
+TEST(nntrainer_Pooling2D, forwarding_04_p) {
+  int status = ML_ERROR_NONE;
+  nntrainer::Pooling2DLayer layer;
+  std::vector<std::string> input_str;
+  nntrainer::TensorDim previous_dim;
+  previous_dim.setTensorDim("1:2:5:5");
+  layer.setInputDimension(previous_dim);
+
+  input_str.push_back("pooling = global_average");
+
+  status = layer.setProperty(input_str);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = layer.initialize(false);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  nntrainer::Tensor in(1, 2, 5, 5);
+  nntrainer::Tensor out;
+  std::ifstream file("test_1_goldenConv2DResult.out");
+  in.read(file);
+  file.close();
+  out = layer.forwarding(in, status);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  float *out_ptr;
+
+  float golden[2] = {6.6994767, 7.1483521};
+
+  out_ptr = out.getData();
+
+  for (int i = 0; i < 2; ++i) {
+    EXPECT_FLOAT_EQ(out_ptr[i], golden[i]);
+  }
+}
+
+/**
+ * @brief Pooling 2D Layer
+ */
+TEST(nntrainer_Pooling2D, forwarding_05_p) {
+  int status = ML_ERROR_NONE;
+  nntrainer::Pooling2DLayer layer;
+  std::vector<std::string> input_str;
+  nntrainer::TensorDim previous_dim;
+  previous_dim.setTensorDim("2:3:5:5");
+  layer.setInputDimension(previous_dim);
+
+  input_str.push_back("pooling_size= 2,2");
+  input_str.push_back("stride=1, 1");
+  input_str.push_back("padding=0,0");
+  input_str.push_back("pooling = global_max");
+
+  status = layer.setProperty(input_str);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = layer.initialize(false);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  nntrainer::Tensor in(2, 3, 5, 5);
+  nntrainer::Tensor out, result(2, 3, 4, 4);
+  std::ifstream file("test_2_goldenConv2DResult.out");
+  in.read(file);
+  file.close();
+  out = layer.forwarding(in, status);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  float *out_ptr;
+  float golden[6] = {9.60282, 8.78552, 9.1152, 9.29397, 8.580175, 8.74109};
+
+  out_ptr = out.getData();
+
+  for (int i = 0; i < 6; ++i) {
+    EXPECT_FLOAT_EQ(out_ptr[i], golden[i]);
+  }
+}
+
+/**
+ * @brief Pooling 2D Layer
+ */
+TEST(nntrainer_Pooling2D, forwarding_06_p) {
```

Comment:
How about using test class for this and use `TEST_F` by gtest? We will have many more tests for pooling to test different configuration of pooling types, strides and padding, and `TEST_F` would make it simple.

Reply:
Updated.